cognite/client/_api/user_profiles.py

```diff
@@ -0,0 +1,154 @@
+
+        Retrieves the user profile of the principal issuing the request, i.e. the principal *this* CogniteClient was instantiated with.
+
+        Returns:
+            UserProfile: Your own user profile.
+
+        Raises:
+            CogniteAPIError: If this principal doesn't have a user profile, you get a not found (404) response code.
+
+        Examples:
+
+            Get your own user profile:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.iam.user_profiles.me()
+        """
+        return UserProfile._load(self._get(self._RESOURCE_PATH + "/me").json())
+
+    def list(self, limit: int | None = LIST_LIMIT_DEFAULT, initial_cursor: str | None = None) -> UserProfileList:
+        """`List user profiles <https://developer.cognite.com/api#tag/User-profiles/paths/~1profiles/get>`_
+
+        List all user profiles in the current CDF project. The results are ordered alphabetically by name.
+
+        Args:
+            limit (int | None): Maximum number of user profiles to return. Defaults to 25. Set to -1, float("inf") or None to return all.
+            initial_cursor (str | None): Start fetching from a specific cursor.
+
+        Returns:
+            UserProfileList: List of user profiles.
+
+        Examples:
+
+            List all user profiles:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.iam.user_profiles.list(limit=None)
+        """
+        return self._list(
+            "GET",
+            list_cls=UserProfileList,
+            resource_cls=UserProfile,
+            limit=limit,
+            initial_cursor=initial_cursor,
+        )
+
+    def retrieve(self, user_identifier: str) -> UserProfile | None:
+        """`Retrieve a single user profile by user identifier. <https://developer.cognite.com/api#tag/User-profiles/paths/~1profiles~1byids/post>`_
+
+        Retrieves one user profile as indexed by the user identifier in the same CDF project.
+
+        Args:
+            user_identifier (str): The user identifier of the user profile to retrieve.
+
+        Returns:
+            UserProfile | None: The requested user profile or None if it doesn't exist.
+
+        Examples:
+
+            Get user profile by user identifier:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.iam.user_profiles.retrieve("foo")
+        """
+        identifier = UserIdentifierSequence.load(user_identifier).as_singleton()
+        return self._retrieve_multiple(
+            list_cls=UserProfileList,
+            resource_cls=UserProfile,
+            identifiers=identifier,
+        )
+
+    def retrieve_multiple(self, user_identifiers: Sequence[str]) -> UserProfileList:
```

Comment:
Suggest using a single method with overloads instead of the split. This is what we have done for all newer implementations in the SDK. 
```python
retrieve(self, user_identifier: str | Sequence[str]) -> UserProfile | UserProfileList | None:
```

Reply:
Sure, I never liked typing `_multiple` anyway 😂